gobblin-config-management/gobblin-config-core/src/main/java/gobblin/config/store/hdfs/SimpleHDFSConfigStore.java

```diff
@@ -0,0 +1,341 @@
+
+  /**
+   * Constructs a {@link SimpleHDFSConfigStore} using a given {@link FileSystem} and a {@link URI} that points to the
+   * physical location of the store root.
+   *
+   * @param fs the {@link FileSystem} the {@link ConfigStore} is stored on.
+   * @param physicalStoreRoot the fully qualified {@link URI} of the store root, the {@link URI#getScheme()} of the
+   *                          {@link URI} should match the {@link FileSystem#getScheme()} of the given {@link FileSystem}.
+   */
+  protected SimpleHDFSConfigStore(FileSystem fs, URI physicalStoreRoot) {
+    Preconditions.checkNotNull(fs, "fs cannot be null!");
+    Preconditions.checkNotNull(physicalStoreRoot, "physicalStoreRoot cannot be null!");
+
+    this.fs = fs;
+
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(physicalStoreRoot.getScheme()),
+        "The physicalStoreRoot must have a valid scheme!");
+    Preconditions.checkArgument(physicalStoreRoot.getScheme().equals(fs.getUri().getScheme()),
+        "The scheme of the physicalStoreRoot and the filesystem must match!");
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(physicalStoreRoot.getPath()),
+        "The path of the physicalStoreRoot must be valid as it is the root of the store!");
+
+    this.physicalStoreRoot = physicalStoreRoot;
+  }
+
+  /**
+   * Returns a {@link String} representation of the highest version stored in the {@link ConfigStore}. This method
+   * simply determines the highest version by doing an {@code ls} on the store root, and returning the first entry
+   * found. Since this method performs an {@code ls}, the highest version should also correspond to the version that
+   * comes first in lexicographical order.
+   *
+   * @return a {@link String} representing the current (highest) version of the {@link ConfigStore}.
+   */
+  @Override
+  public String getCurrentVersion() {
+    Path configStoreDir = new Path(new Path(this.physicalStoreRoot), CONFIG_STORE_NAME);
+
+    try {
+      if (!this.fs.exists(configStoreDir)) {
+        throw new VersionDoesNotExistException(getStoreURI(), "CURRENT", "");
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(
+          String.format("Error while checking if the configStoreDir: \"%s\" exists", configStoreDir), e);
+    }
+
+    try {
+      for (FileStatus status : this.fs.listStatus(configStoreDir)) {
+        return status.getPath().getName();
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(
+          String.format("Error while checking current version for configStoreDir: \"%s\"", configStoreDir), e);
+    }
+    return null;
+  }
+
+  /**
+   * Returns a {@link URI} representing the physical store {@link URI} where the {@link URI#getPath()} is the path to
+   * the root of the {@link ConfigStore}.
+   *
+   * @return a {@link URI} representing the physical store {@link URI} (e.g. hdfs://[authority]:[port][path-to-root]).
+   */
+  @Override
+  public URI getStoreURI() {
```

Comment:
I believe @ibuenros brought up a similar comment. Let's discuss offline. I can change it to the logical URI, but the JavaDocs in `ConfigStore` say it should return the physical URI.